src/AWS.Deploy.ServerMode.Client/RestAPIClientFactory.cs

```diff
@@ -0,0 +1,30 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.\r
+// SPDX-License-Identifier: Apache-2.0
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Amazon.Runtime;
+
+namespace AWS.Deploy.ServerMode.Client
+{
+    public static class RestAPIClientFactory
+    {
+        /// <summary>
+        /// Factory method that runs the deployment tool in server mode
+        /// and creates the <see cref="RestAPIClient"/> client with the server mode start configuration.
+        /// TODO: ability to pass configuration file path
+        /// </summary>
+        /// <param name="credentialsGenerator">Func to that provides AWS credentials</param>
+        /// <param name="cancellationToken">A cancellation token that can be used by other objects or threads to receive notice of cancellation.</param>
+        /// <returns>Returns <see cref="RestAPIClient"/> client for the current session.</returns>
+        public static async Task<IRestAPIClient> BuildAPIClient(Func<Task<AWSCredentials>> credentialsGenerator, CancellationToken cancellationToken)
```

Comment:
From the Toolkit, I plan to request different clients from the running session. Each client will have its own credentials generator. What do you think about a `ServerModeStartInfo` extension method that performs the construction of the http client and the rest client, and returns the rest client? This would make a nice abstraction that lets a caller get the rest client without worrying about the details.

Reply:
Sounds great. I love the interface approach as well. In addition, I'd like if it was possible to:
* query if a server is up (this will let the Toolkit know if the server stops that it should update the UI accordingly and discard its RestApiClient)
* stop a running server (this will let the Toolkit attempt to clean up running server on shut down)

Reply:
I'd like to be able to get a rest api client in a non-async manner. Would it be possible to add a version of GetRestAPIClient that doesn't try to start the server but just produces the api client?

Reply:
I would prefer having the code of restarting the server at one place. Either the caller, make sure server is running, if not first call Start and then get the client. Or let the Session handle this. Both ways, it works for me.

Reply:
> I'd like to be able to get a rest api client in a non-async manner.

This will be tricky, we have to take out the Start server out of the call stack then,

Reply:
Ok, lets have the caller (Toolkit) take responsibility for re/starting the server. This way it has the flexibility to handle the lifecycle as needed.

Reply:
Updated.